### PR TITLE
[op-conductor] part 1 - conductor main control loop

### PR DESCRIPTION
**Pull request by 0x00101010**
**Description**

* Implemented high level control loop structure, left actual control logic unimplemented, will do them in a follow up PR.
* Generated mocks for Consensus, SequencerControl and HealthMonitor

**Tests**

Added tests to make sure that conductor initialization / pause / resume / start / stop works as intended.

**Metadata**

- https://github.com/ethereum-optimism/protocol-quest/issues/47

### CODE REVIEW

File: op-conductor/conductor/service_test.go [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	s.healthUpdateCh = make(chan bool)
+	s.hmon.EXPECT().Start().Return(nil)
+	s.hmon.EXPECT().Subscribe().Return(s.healthUpdateCh)
+
+	s.leaderUpdateCh = make(chan bool)
+	s.cons.EXPECT().LeaderCh().Return(s.leaderUpdateCh)
+
+	err = s.conductor.Start(s.ctx)
+	s.NoError(err)
+	s.False(s.conductor.Stopped())
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply op-conductor/conductor/service_test.go:115]: incorrect suggestion

File: op-conductor/conductor/service_test.go [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	err = s.conductor.Resume(s.ctx)
+	s.NoError(err)
+	s.False(s.conductor.Paused())
+
+	// Stop
+	s.hmon.EXPECT().Stop().Return(nil)
+	s.cons.EXPECT().Shutdown().Return(nil)
+	err = s.conductor.Stop(s.ctx)
+	s.NoError(err)
+	s.True(s.conductor.Stopped())
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply op-conductor/conductor/service_test.go:138]: This is not needed as `s.healthUpdateCh <- true` blocks if not consumed.

File: op-conductor/conductor/service_test.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+	// Resume
+	err = s.conductor.Resume(s.ctx)
+	s.NoError(err)
+	s.False(s.conductor.Paused())
+
+	// Resume
+	err = s.conductor.Resume(s.ctx)
+	s.NoError(err)
+	s.False(s.conductor.Paused())
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply op-conductor/conductor/service_test.go:161]: There is checks to handle that.

File: op-conductor/conductor/service_test.go [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+			ChannelTimeout:          300,
+			L1ChainID:               big.NewInt(1),
+			L2ChainID:               big.NewInt(2),
+			CanyonTime:              &now,
+			BatchInboxAddress:       [20]byte{1, 2},
+			DepositContractAddress:  [20]byte{2, 3},
+			L1SystemConfigAddress:   [20]byte{3, 4},
+			ProtocolVersionsAddress: [20]byte{4, 5},
+		},
+	}
```
[the remark this answers is not in the document]
Comment (0x00101010) [reply op-conductor/conductor/service_test.go:69]: I don't believe this is needed as it should be cleaned up automatically by CI.

File: op-conductor/client/mocks/SequencerControl.go [hunk 5]
```diff
@@ -0,0 +1,287 @@
+// Code generated by mockery v2.28.1. DO NOT EDIT.
```
Comment (mslipper) [inline op-conductor/client/mocks/SequencerControl.go:1]: [nit] should be sequencer_control.go

File: op-conductor/health/mocks/HealthMonitor.go [hunk 6]
```diff
@@ -0,0 +1,158 @@
+// Code generated by mockery v2.28.1. DO NOT EDIT.
```
Comment (mslipper) [inline op-conductor/health/mocks/HealthMonitor.go:1]: [nit] health_monitor.go

File: op-conductor/consensus/mocks/Consensus.go [hunk 7]
```diff
@@ -0,0 +1,539 @@
+// Code generated by mockery v2.28.1. DO NOT EDIT.
```
Comment (mslipper) [inline op-conductor/consensus/mocks/Consensus.go:1]: [nit] consensus.go

**Review by mslipper — APPROVED** [review]

File: op-conductor/consensus/mocks/Consensus.go (hunk 7 above)
Comment (0x00101010) [reply op-conductor/consensus/mocks/Consensus.go:1]: These names seem to be standard behavior from mockery, and is consistent with how other mocks are generated within the repo, example:

* https://github.com/ethereum-optimism/optimism/blob/develop/op-node/p2p/gating/mocks/ExpiryStore.go
* https://github.com/ethereum-optimism/optimism/blob/develop/op-service/txmgr/mocks/TxManager.go

That said, happy to try to change all to snake case if necessary.

If not, please help add it to merge queue

**Review by zhwrd — APPROVED** [review]

**Pull request merged**